compiler/deSugar/Match.hs

```diff
@@ -839,6 +839,12 @@ matchSinglePat (Var var) ctx pat ty match_result
 
 matchSinglePat scrut hs_ctx pat ty match_result
   = do { var           <- selectSimpleMatchVarL Many pat
+                            -- Currently called only from places where an
+                            -- case-Many is expected. This is not a robust
+                            -- property, in fact it may not even be true today!
```

Comment:
This is rather confusing: in the first sentence we say that it's called only when case-Many is expected, in the second sentence we say this may not even be true.

Reply:
Yeah, it's probably a bug in the making. That's the best phrasing I could find, but suggestions welcome.